react-app/src/api/govAPI.ts

```diff
@@ -14,20 +16,100 @@ import {
 } from "../utils/coin";
 import { SignedTx, useCosmosAPI } from "./cosmosAPI";
 
+interface DepositParams {
+  minDeposit: BigNumberCoin;
+  /**
+   * Maximum deposit period in days
+   */
+  maxDepositPeriod: number;
+}
+
+/**
+ * TallyParams defines the params for tallying votes on governance proposals.
+ * results are returned in percentages
+ */
+export interface TallyParams {
+  /**
+   * Minimum percentage of total stake needed to vote for a result to be
+   *  considered valid.
+   */
+  quorum: number;
+  /** Minimum proportion of Yes votes for proposal to pass. Default value: 0.5. */
+  threshold: number;
+  /**
+   * Minimum value of Veto votes to Total votes ratio for proposal to be
+   *  vetoed. Default value: 1/3.
+   */
+  vetoThreshold: number;
+}
+
+interface VotingParams {
+  /**
+   * Maximum voting period in days
+   */
+  votingPeriod: number;
+}
+
+export interface GovParams {
+  deposit: DepositParams;
+  tally: TallyParams;
+  voting: VotingParams;
+}
+
 interface IGovAPI {
-  getMinDepositParams(): Promise<BigNumberCoin>;
   signSubmitProposalTx(
     proposal: ProposalContentBody,
     initialDeposit: BigNumber,
     memo?: string
   ): Promise<SignedTx>;
+  getMinDepositParams(): Promise<BigNumberCoin>;
+  getAllParams(): Promise<GovParams>;
 }
 
+const CoinMinimalDenom = Config.chainInfo.currency.coinMinimalDenom;
+const CoinDenom = Config.chainInfo.currency.coinDenom;
+
+/**
+ * Convert cosmjs Duration to days
+ * return 0 if duration is undefined
+ */
+const convertDurationToDays = (duration: Duration | undefined) => {
+  if (!duration) {
+    return 0;
+  }
+  return duration.seconds.div(60 * 60 * 24).toInt();
+};
+
+const decodeUint8ArrayToBigNumberPercentage = (n: Uint8Array) => {
+  return new BigNumber("0." + Buffer.from(n).toString("utf8"))
+    .times(100)
+    .toNumber();
+};
```

Comment:
i think can do this in UI component, for coins I think can convert in API level but for data/params I think should keep them in their base unit as they could be used in different ways and contexts

Reply:
also is there other ways to do this instead of adding `0.` in front :sosad:?

Reply:
I can't think of a better way...

Reply:
seems they are using the Dec type for these params
https://github.com/cosmos/cosmos-sdk/blob/d0043914ba7c37c3a0d7039d2c2a2aca6b38a93b/x/gov/types/gov.pb.go#L453-L462

Can we try to use `Decimal` from `@cosmjs/math`?